yt_dlp/extractor/nebula.py

```diff
@@ -179,60 +113,171 @@ def _fetch_zype_access_token(self, video_id):
+        },
+    ]
 
-        # try to authenticate using credentials if no valid token has been found
-        if not self._nebula_token:
-            self._nebula_token = self._retrieve_nebula_auth()
+    def _fetch_video_metadata(self, slug):
+        return self._call_nebula_api(f'https://content.watchnebula.com/video/{slug}/',
+                                     video_id=slug,
+                                     auth_type='bearer',
+                                     note='Fetching video meta data')
 
     def _real_extract(self, url):
-        display_id = self._match_id(url)
-        api_key = self._retrieve_zype_api_key(url, display_id)
+        slug = self._match_id(url)
+        video = self._fetch_video_metadata(slug)
+        return self._build_video_info(video)
 
-        response = self._call_zype_api('/videos', {'friendly_title': display_id},
-                                       display_id, api_key, note='Retrieving metadata from Zype')
-        if len(response.get('response') or []) != 1:
-            raise ExtractorError('Unable to find video on Zype API')
-        video_meta = response['response'][0]
 
-        video_id = video_meta['_id']
-        zype_access_token = self._fetch_zype_access_token(display_id)
+class NebulaCollectionIE(NebulaBaseIE):
+    IE_NAME = 'nebula:collection'
+    _VALID_URL = r'https?://(?:www\.)?(?:watchnebula\.com|nebula\.app)/(?!videos/)(?P<id>[-\w]+)'
+    _TESTS = [
+        {
+            'url': 'https://nebula.app/tom-scott-presents-money',
+            'info_dict': {
+                'id': 'tom-scott-presents-money',
+                'title': 'Tom Scott Presents: Money',
+                'description': 'Tom Scott hosts a series all about trust, negotiation and money.',
+            },
+            'playlist_count': 5,
+            'params': {
+                'usenetrc': True,
+            },
+        }, {
+            'url': 'https://nebula.app/lindsayellis',
+            'info_dict': {
+                'id': 'lindsayellis',
+                'title': 'Lindsay Ellis',
+                'description': 'Enjoy these hottest of takes on Disney, Transformers, and Musicals.',
+            },
+            'playlist_mincount': 100,
+            'params': {
+                'usenetrc': True,
+            },
+        },
+    ]
 
-        channel_title = self._extract_channel_title(video_meta)
+    def _generate_playlist_entries(self, collection_id, channel):
+        page_nr = 1
+        episodes = channel['episodes']['results']
+        while episodes:
+            for episode in episodes:
+                yield self._build_video_info(episode)
+            next_url = channel['episodes']['next']
+            if next_url:
+                page_nr += 1
+                channel = self._call_nebula_api(next_url, collection_id, auth_type='bearer', note=f'Retrieving channel page {page_nr}')
+                episodes = channel['episodes']['results']
+            else:
+                episodes = None
```

Comment:
Oh, this actually turned out much simpler than I initially thought. I didn't think to extract one page in `_real_extract` and then pass that data inside 👍 

I would write this function like this, but your implementation is fine as well. So I won't insist

```py
        episodes = channel['episodes']['results']
        for page_num in itertools.count(1):
            for episode in episodes:
                yield self._build_video_info(episode)
            next_url = channel['episodes']['next']
            if not next_url:
                break
            episodes = self._call_nebula_api(
                next_url, collection_id, auth_type='bearer',
                note=f'Retrieving channel page {page_num}')['episodes']['results']
```

Reply:
Ha, that's pretty similar to the version I had first, but then I thought you would mind the explicit control flow. 😉

I think this is good. I did however change the count back to start at 2, because the first page of episodes has already been fetched alongside the channel metadata. This is also consistent with the backend behavior, where the very first `next_url` already contains a `page=2` query param.